src/pages/iou/steps/MoneyRequestConfirmPage.js

```diff
@@ -89,14 +90,14 @@ function MoneyRequestConfirmPage(props) {
             IOU.resetMoneyRequestInfo(moneyRequestId);
         }
 
-        if (_.isEmpty(props.iou.participants) || props.iou.amount === 0 || shouldReset) {
+        if (_.isEmpty(props.iou.participants) || (props.iou.amount === 0 && !props.iou.receiptPath) || shouldReset) {
```

Comment:
This logic must be applied to other similar pages as well e.g. `MoneyRequestDescriptionPage`. This shall fix this bug https://github.com/Expensify/App/pull/23046#pullrequestreview-1542223790